pkg/apis/pingcap/v1alpha1/backup.go

```diff
@@ -316,18 +313,60 @@ func NeedNotClean(backup *Backup) bool {
 }
 
 // ParseLogBackupSubcommand parse the log backup subcommand from cr.
-// The parse priority of the command is stop > truncate > start.
 func ParseLogBackupSubcommand(backup *Backup) LogSubCommandType {
 	if backup.Spec.Mode != BackupModeLog {
 		return ""
 	}
+
+	// Compatible with the old version
 	if backup.Spec.LogStop {
+		if IsLogBackupAlreadyStop(backup) && backup.Spec.LogTruncateUntil != "" && backup.Spec.LogTruncateUntil != backup.Status.LogSuccessTruncateUntil {
+			return LogTruncateCommand
+		}
 		return LogStopCommand
 	}
-	if backup.Spec.LogTruncateUntil != "" {
+
+	var subCommand LogSubCommandType
+	switch backup.Spec.LogSubcommand {
+	case "":
+		fallthrough
+	case LogStartCommand:
+		if IsLogBackupAlreadyPaused(backup) {
+			subCommand = LogResumeCommand
+		} else {
+			subCommand = LogStartCommand
+		}
+	case LogStopCommand:
+		subCommand = LogStopCommand
+	case LogPauseCommand:
+		subCommand = LogPauseCommand
+	default:
+		return LogUnknownCommand
```

Comment:
if set `LogSubcommand` as `""`, what the subcommand will be used, `log-unkown` or `log-start`.

Reply:
Let me make it more clear:

1. 
Actually we don't need the "" as a subcommand. The only reason we keep it is for compatible with the old version interface.
The old usage is demonstrated here: https://github.com/pingcap/tidb-operator/pull/4682
TL;DR
In the old version, we use this CRD to start a new task:

```
Spec:
# empty field
```

Therefore, `LogSubcommand == ""` means `LogStartCommand`

2.
I add `LogUnknownCommand` for handling unexpected input from users, but now we use `+kubebuilder:validation:Enum` instead. This field is useless now. I keep it just for a double insurance.

3.
Now I refactored the code like

```
if backup.Spec.LogSubcommand == "" {
        if backup.Spec.LogStop || IsLogBackupAlreadyStop(backup) {
            subCommand = LogStopCommand
        } else if IsLogBackupAlreadyPaused(backup){
            subCommand = LogResumeCommand
        } else {
			subCommand = LogStartCommand
		}
    } else {
        switch backup.Spec.LogSubcommand {
        case LogStartCommand:
            if IsLogBackupAlreadyPaused(backup) {
                subCommand = LogResumeCommand
            } else {
                subCommand = LogStartCommand
            }
        case LogStopCommand:
            subCommand = LogStopCommand
        case LogPauseCommand:
            subCommand = LogPauseCommand
        default:
            return LogUnknownCommand
        }
    }
```
I suppose it could be more clear now.